packages/ponder-metadata-api/src/types/api.ts

```diff
@@ -27,7 +27,7 @@ export interface PonderMetadataMiddlewareOptions<AppInfo, EnvVars extends Ponder
   };
 
   /** Public clients for each blockchain network fetching data */
-  publicClients: Record<number, PublicClient>;
+  publicClients: Record<string, PublicClient>;
```

Comment:
Changing record key type to string to follow idiomatic JS coding style.

Valibot expects the keys of a record to be of string type. Object.entires and Object.keys would also include keys as string types. Serialized object has its numeric keys formatted as strings.